### PR TITLE
docs(how-to-guides): add ros2_graph and ros2doctor to debug guidelines

### PR DESCRIPTION
**Pull request by xmfcx**
Signed-off-by: M. Fatih Cırıt <mfc@leodrive.ai>

## Description

Adding some simple but useful debug tips.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The Reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: docs/how-to-guides/debug-autoware.md [hunk 1]
```diff
@@ -63,6 +63,52 @@ ros2 run rqt_console rqt_console
    ros2 run rqt_plot rqt_plot
    ```
 
+### Using ros2_graph
+
+[`ros2_graph`](https://github.com/kiwicampus/ros2_graph) can be used to generate [mermaid](https://mermaid.js.org/#/) description of ROS2 graphs to add on your markdown files.
```
Comment (kenji-miyake) [inline docs/how-to-guides/debug-autoware.md:68]: We need a space.
https://discourse.ros.org/t/should-i-put-a-space-between-ros-and-2/27949

```suggestion
[`ros2_graph`](https://github.com/kiwicampus/ros2_graph) can be used to generate [mermaid](https://mermaid.js.org/#/) description of ROS 2 graphs to add on your markdown files.
```

File: docs/how-to-guides/debug-autoware.md [hunk 2]
```diff
@@ -63,6 +63,52 @@ ros2 run rqt_console rqt_console
...
+
+# or multiple nodes
+ros2_graph /turtlesim /teleop_turtle
+```
+
+You can then visualize these graphs with:
+
+- [mermaid live editor](https://mermaid-js.github.io/mermaid-live-editor/)
+- Visual studio code extension [mermaid preview](https://marketplace.visualstudio.com/items?itemName=vstirbu.vscode-mermaid-preview)
+- Jetbrains IDEs [with native support](https://www.jetbrains.com/go/guide/tips/mermaid-js-support-in-markdown/)
```
Comment (kenji-miyake) [inline docs/how-to-guides/debug-autoware.md:94]: style
https://en.wikipedia.org/wiki/JetBrains

```suggestion
- JetBrains IDEs [with native support](https://www.jetbrains.com/go/guide/tips/mermaid-js-support-in-markdown/)
```

File: docs/how-to-guides/debug-autoware.md [hunk 3]
```diff
@@ -63,6 +63,52 @@ ros2 run rqt_console rqt_console
...
+ros2_graph /turtlesim -o turtle_diagram.md
+
+# or multiple nodes
+ros2_graph /turtlesim /teleop_turtle
+```
+
+You can then visualize these graphs with:
+
+- [mermaid live editor](https://mermaid-js.github.io/mermaid-live-editor/)
+- Visual studio code extension [mermaid preview](https://marketplace.visualstudio.com/items?itemName=vstirbu.vscode-mermaid-preview)
```
Comment (kenji-miyake) [inline docs/how-to-guides/debug-autoware.md:93]: [style] I think using exact expressions is better.

```suggestion
- [Mermaid Live Editor](https://mermaid-js.github.io/mermaid-live-editor/)
- Visual Studio Code extension [mermaid preview](https://marketplace.visualstudio.com/items?itemName=vstirbu.vscode-mermaid-preview)
```

File: docs/how-to-guides/debug-autoware.md [hunk 4]
```diff
@@ -63,6 +63,52 @@ ros2 run rqt_console rqt_console
...
+
+`ros2doctor` checks all aspects of ROS 2, including platform, version, network, environment, running systems and more, and warns you about possible errors and reasons for issues.
+
+It's as simple as just running `ros2 doctor` in your terminal.
+
+It has the ability to list "Subscribers without publishers" for all topics in the system.
+
+And this information can help you find if a necessary node isn't running.
+
+For more details, please see the following official documentation:
```
Comment (kenji-miyake) [inline docs/how-to-guides/debug-autoware.md:108]: [style] We usually don't need `please`.
https://developers.google.com/style/tone#politeness-and-use-of-please
```suggestion
For more details, see the following official documentation:
```

File: docs/how-to-guides/debug-autoware.md [hunk 5]
```diff
@@ -63,6 +63,52 @@ ros2 run rqt_console rqt_console
...
+
+- <https://docs.ros.org/en/rolling/Tutorials/Beginner-Client-Libraries/Getting-Started-With-Ros2doctor.html>
```
Comment (kenji-miyake) [inline docs/how-to-guides/debug-autoware.md:110]: I personally this is acceptable, but FYI, there is also a guidelines page about link texts.

https://developers.google.com/style/link-text

File: docs/how-to-guides/debug-autoware.md [hunk 6]
```diff
@@ -63,6 +63,52 @@ ros2 run rqt_console rqt_console
...
+
+It's as simple as just running `ros2 doctor` in your terminal.
+
+It has the ability to list "Subscribers without publishers" for all topics in the system.
+
+And this information can help you find if a necessary node isn't running.
+
+For more details, see the following official documentation:
+
+- <https://docs.ros.org/en/rolling/Tutorials/Beginner-Client-Libraries/Getting-Started-With-Ros2doctor.html>
```
Comment (xmfcx) [inline docs/how-to-guides/debug-autoware.md:110]:
```suggestion
For more details, see the following official documentation for [Using ros2doctor to identify issues](https://docs.ros.org/en/rolling/Tutorials/Beginner-Client-Libraries/Getting-Started-With-Ros2doctor.html)
```

File: docs/how-to-guides/debug-autoware.md (hunk 5 above)
Comment (xmfcx) [reply docs/how-to-guides/debug-autoware.md]: I see, I've applied https://github.com/autowarefoundation/autoware-documentation/pull/316/commits/0757f7bb7981bd39b2a48bee7bd82e0486d561f0 to address this.

**Comment by xmfcx** [comment]
@kenji-miyake thanks for the detailed review!

File: docs/how-to-guides/debug-autoware.md [hunk 7]
```diff
@@ -63,6 +63,50 @@ ros2 run rqt_console rqt_console
...
+
+`ros2doctor` checks all aspects of ROS 2, including platform, version, network, environment, running systems and more, and warns you about possible errors and reasons for issues.
+
+It's as simple as just running `ros2 doctor` in your terminal.
+
+It has the ability to list "Subscribers without publishers" for all topics in the system.
+
+And this information can help you find if a necessary node isn't running.
+
+For more details, see the following official documentation for [Using ros2doctor to identify issues](https://docs.ros.org/en/rolling/Tutorials/Beginner-Client-Libraries/Getting-Started-With-Ros2doctor.html)
```
Comment (kenji-miyake) [inline docs/how-to-guides/debug-autoware.md:108]: Need a dot?

```suggestion
For more details, see the following official documentation for [Using ros2doctor to identify issues](https://docs.ros.org/en/rolling/Tutorials/Beginner-Client-Libraries/Getting-Started-With-Ros2doctor.html).
```

Comment (xmfcx) [reply docs/how-to-guides/debug-autoware.md:108]: not so important but why not :D

**Pull request merged**